core/src/main/java/de/javakaffee/web/msm/storage/RedisStorageClient.java

```diff
@@ -0,0 +1,292 @@
+/*
+ * Copyright 2016 Markus Ellinger
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an &quot;AS IS&quot; BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package de.javakaffee.web.msm.storage;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.ObjectInput;
+import java.io.ObjectInputStream;
+import java.io.ObjectOutput;
+import java.io.ObjectOutputStream;
+import java.util.concurrent.Callable;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeoutException;
+import java.util.concurrent.TimeUnit;
+import java.util.ArrayList;
+import java.util.List;
+import java.nio.ByteBuffer;
+import java.nio.charset.Charset;
+import java.nio.charset.StandardCharsets;
+
+import org.apache.juli.logging.Log;
+import org.apache.juli.logging.LogFactory;
+
+import redis.clients.jedis.BinaryJedis;
+import redis.clients.jedis.exceptions.JedisConnectionException;
+
+import de.javakaffee.web.msm.NamedThreadFactory;
+
+/**
+ * Storage client backed by a Jedis client instance.
+ */
+public class RedisStorageClient implements StorageClient {
+    protected static final Log _log = LogFactory.getLog(RedisStorageClient.class);
+
+    private final String _host;
+    private final int _port;
+    private final boolean _ssl;
+    private final JedisPool _pool = new JedisPool();
+    private final ExecutorService _executor = Executors.newFixedThreadPool(
+            Runtime.getRuntime().availableProcessors(), new NamedThreadFactory("msm-redis-client"));
```

Comment:
The work here should be IO bound and not cpu bound. Therefore the max number of threads could be higher, perhaps a cached thread pool could be used alternatively.